xdk/build.gradle.kts

```diff
@@ -1,602 +1,408 @@
+    // This is file, so it can be resolved at the top level.
+    val localDist = xdkBuildLogicProvider.get().resolveLocalXdkInstallation()
+    val dest = xdkDist.getLocalDistBackupDir(localDist.name)
+    from(localDist) {
+        into(dest)
+    }
+    doLast {
+        logger.lifecycle("$prefix Backed up local installation at $localDist to ${dest.get()}")
     }
 }
 
-val distZIP = tasks.register<Zip>("distZIP") {
-    group       = "Distribution"
-    description = "Create the XDK .zip file"
-
-    dependsOn(build)
-    dependsOn(prepareDirs)
-
-    archiveFileName.set("xdk-${distName}.zip")
-    destinationDirectory.set(file("$distDir/"))
-    from("$buildDir/") {
-        include("xdk/**")
+val purgeLocalDist by tasks.registering {
+    group = DISTRIBUTION_TASK_GROUP
+    description = "Remove any existing local XDK distribution."
+    dependsOn(backupLocalDist)
+    mustRunAfter(backupLocalDist)
+    doLast {
+        val localDistDir = findLocalXdkInstallation()
+        val allowOverwrite = getXdkPropertyBoolean("org.xtclang.build.allowOverwriteLocalDist", true)
+        if (localDistDir != null) {
+            if (!allowOverwrite) {
+                logger.lifecycle("$prefix '$name' will purge and replace existing local installation at: '$localDistDir'.")
+                delete(localDistDir)
+                mkdir(localDistDir)
+            } else {
+                logger.lifecycle("$prefix '$name' will overwrite existing local installation at '$localDistDir'.")
+            }
+        }
     }
 }
 
-val distEXE = tasks.register("distEXE") {
-    group = "Distribution"
-    description = "Create the XDK .exe file (Windows installer)"
+/**
+ * Overwrite or install a local distribution of the XDK.
+ *
+ * TODO: This should be the only task that triggers the buildRepo plugin. Can we do that without
+ *   knowing if this should be run or not, avoiding configuration of the extra publication if
+ *   task is not set to run?
+ *
+ * TODO: It is unfortunate that we don't copy the entire XDK distribution over the existing
+ *   install, because it creates confusion about dependencies. Ideally, the entire local install
+ *   should be overwritten with a complete self-contained new distribution, but this is not the
+ *   case in "master", and we first implement it exactly like that, semantically, to make sure
+ *   nothing breaks in existing use cases. All in all, it's rather horrible to go in and replace
+ *   parts of an existing home brew (or other package manager) based installation, and should
+ *   really never be allowed, as it leads to bit skew.
+ *
+ * TODO: @aalmiray recommends application plugin run script generation, and that makes sense to me.
+ *   It should be possible to hook up JReleaser and/or something like launch4j to cross compile
+ *   binary launchers for different platforms, if that is what we want instead of the current
+ *   solution.
+ */
+val installLocalDist by tasks.registering {
```

Comment:
This task blew up - I posted the error to the Discord

Reply:
Do you have a link to a build scan on how it blew up?

Reply:
I had a file resolution issue remaining. I'll add a check that this works to the GitHub CI runner when I have time.

Reply:
I did the command that you recommended, `./gradlew build --scan --stacktrace`, but it looked like it blew up before producing anything else but the output I posted to Discord.